crates/primitives/src/transaction.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{AccessListItem, Address, Bytes, TxKind, B256, GAS_PER_BLOB, U256};
+
+/// Trait for retrieving transaction information required for execution.
+pub trait Transaction {
+    /// Caller aka Author aka transaction signer.
+    fn caller(&self) -> &Address;
```

Comment:
I used the philosophy that anything larger than a `u64` (pointer) should use a `&` whereas anything smaller or equal to a `u64` is returned by value. This is also the guideline for when to derive `Copy` or not, as it avoids unnecessarily large copies.

I'm not sure whether the presence of lifetimes in the return arguments might be a concern for some people though.